Code/RDGeneral/testRDTags.cpp

```diff
@@ -0,0 +1,149 @@
+#include "tags.h"
+#include "Invariant.h"
+
+using namespace RDKit;
+
+template < typename T > std::string to_string( const T& n )
+{
+  std::ostringstream stm ;
+  stm << n ;
+  return stm.str() ;
+}
+
+const int MOD=10000;
+const int COUNT=5000000;
+static std::vector<std::string> nums;
+
+void testTags() {
+  for( int idx=0; idx<MOD; ++idx) {
+    nums.push_back( to_string(idx) );
+  }
+
+
+  RDTags tags;
+  {
+    std::clock_t clock1 = std::clock();
+
+    for( int idx=0; idx<COUNT; ++idx) {
+      int i = idx % MOD;
+      CHECK_INVARIANT(tags.get(nums[i]) == common_properties::MAX+i+1,
+                      "Bad map");
+    }
+    std::clock_t clock2 = std::clock();
+    
+    std::cout << "tag gets: " << (double)(clock2-clock1)/CLOCKS_PER_SEC << " s" << std::endl;
+
+  }
+  {
+    std::clock_t clock1 = std::clock();
+
+    for( int idx=0; idx<COUNT; ++idx) {
+      int i = idx % MOD;
+      CHECK_INVARIANT(tags.get(nums[i]) == common_properties::MAX+i+1,
+                      "Bad map");
+    }
+    std::clock_t clock2 = std::clock();
+    
```

Comment:
What was the performance difference here between threadsafe RDTags and non-threadsafe RDTags? Curious, I wouldn't expect too much.